src/main/java/org/apache/commons/lang3/concurrent/AtomicSafeInitializer.java

```diff
@@ -73,12 +72,31 @@
     public final T get() throws ConcurrentException {
         T result;
 
-        while ((result = reference.get()) == null) {
-            if (factory.compareAndSet(null, this)) {
-                reference.set(initialize());
+        synchronized ( reference ) {
```

Comment:
I dont think this is a good idea to make a lock free code use such a large synchronized block now.